commit fea15ab1c61ff626da4ccce951ec1f9291720e6b (refs/changes/76/4576/2)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2018-07-26T21:48:15+00:00 (1 year, 2 months ago)
    
    MORAY-484 Make "minSpareConnections" default to maximum number of connections
    TRITON-620 Moray's connection pool fubar after Manatee cluster state changes

diff --git a/lib/pg.js b/lib/pg.js
index f3a43b7..0bca822 100644
--- a/lib/pg.js
+++ b/lib/pg.js
@@ -969,11 +969,20 @@ module.exports = {
         number('checkInterval', 60000);
         number('connectTimeout', 1000);
         number('maxConnections', 5);
-        number('minSpareConnections', 2);
         number('queryTimeout', 0);
         number('maxQueueLength', 2000);
         number('targetClaimDelay', 500);
 
+        /*
+         * If minSpareConnections hasn't been specified, then we set it to our
+         * maxConnections. Postgres connections are expensive timewise on the
+         * client's side, and resourcewise on the server's side, so this value
+         * should normally not be set to a small value, since it makes it hard
+         * for Moray to then scale up its connections when they're finally
+         * needed.
+         */
+        number('minSpareConnections', opts.maxConnections);
+
         return (new PGPool(opts));
     },
 
diff --git a/package.json b/package.json
index edebab2..9e5bc23 100644
--- a/package.json
+++ b/package.json
@@ -13,7 +13,7 @@
         "bunyan-syslog": "0.2.2",
         "clone": "0.1.11",
         "crc": "0.2.1",
-        "cueball": "2.7.0",
+        "cueball": "~2.7.1",
         "dtrace-provider": "~0.8",
         "deep-equal": "0.0.0",
         "fast": "2.3.2",
diff --git a/sapi_manifests/moray/template b/sapi_manifests/moray/template
index b044a3c..961538f 100644
--- a/sapi_manifests/moray/template
+++ b/sapi_manifests/moray/template
@@ -22,6 +22,9 @@
       "connectTimeout": 4000,
       "checkInterval": 90000,
       "maxConnections": {{MORAY_MAX_PG_CONNS}},
+{{#MORAY_MIN_PG_CONNS}}
+      "minSpareConnections": {{MORAY_MIN_PG_CONNS}},
+{{/MORAY_MIN_PG_CONNS}}
 {{#MORAY_MAX_QUEUE_LENGTH}}
       "maxQueueLength": {{MORAY_MAX_QUEUE_LENGTH}},
 {{/MORAY_MAX_QUEUE_LENGTH}}
diff --git a/sdc/sapi_manifests/moray/template b/sdc/sapi_manifests/moray/template
index b979bc7..a4e9146 100644
--- a/sdc/sapi_manifests/moray/template
+++ b/sdc/sapi_manifests/moray/template
@@ -22,9 +22,15 @@
       "connectTimeout": 4000,
       "checkInterval": 90000,
       "maxConnections": {{MORAY_MAX_PG_CONNS}},
+{{#MORAY_MIN_PG_CONNS}}
+      "minSpareConnections": {{MORAY_MIN_PG_CONNS}},
+{{/MORAY_MIN_PG_CONNS}}
 {{#MORAY_MAX_QUEUE_LENGTH}}
       "maxQueueLength": {{MORAY_MAX_QUEUE_LENGTH}},
 {{/MORAY_MAX_QUEUE_LENGTH}}
+{{#MORAY_TARGET_CLAIM_DELAY}}
+      "targetClaimDelay": {{MORAY_TARGET_CLAIM_DELAY}},
+{{/MORAY_TARGET_CLAIM_DELAY}}
       "maxIdleTime": 270000,
       "user": "moray"
     }
